File: cases.py
import re

def lower_first(word):
    return word[0].lower() + word[1:]

def upper_first(word):
    return word[0].upper() + word[1:]

def is_snake_case(name):
    return name.find('_') != -1

def to_snake_case(name):
    return re.sub("([A-Z])", "_\\1", name).lower()

def to_camel_case(name):
    return lower_first("".join(upper_first(part) for part in name.split('_')))

while (True):
    name = input()
    if(is_snake_case(name)):
        print(to_camel_case(name))
    else:
        print(to_snake_case(name))

File: Cases.cs
using System.Text.RegularExpressions;

while (true) {
    var name = Console.ReadLine();
    if (name.IsSnakeCase()) {
        Console.WriteLine(name.ToCamelCase());
    }
    else {
        Console.WriteLine(name.ToSnakeCase());
    }
}

static class StringExtensions {
    public static string ToLowerFirst(this string word) {
        return word[0].ToString().ToLower() + word.Substring(1);
    }

    public static string ToUpperFirst(this string word) {
        return word[0].ToString().ToUpper() + word.Substring(1);
    }

    public static bool IsSnakeCase(this string name) => name.Contains('_');
    
    public static string ToSnakeCase(this string name) {
        return Regex.Replace(name, "([A-Z])", "_$1").ToLower();
    }
    
    public static string ToCamelCase(this string name) {
        return string.Join("", name.Split('_').Select(part => part.ToUpperFirst())).ToLowerFirst();
    }
}